File: array-hash/14-Longest-Common-Prefix.go
func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}

	prefix := strs[0]
	for i := 1; i < len(strs); i++ {
		for len(strs[i]) < len(prefix) {
			prefix = prefix[:len(prefix)-1]
		}

		for strs[i][:len(prefix)] != prefix {
			prefix = prefix[:len(prefix)-1]
			if prefix == "" {
				return ""
			}
		}
	}
	return prefix
}

File: reference/array-hash/14-Longest-Common-Prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""

        prefix = strs[0]
        for i in range(1, len(strs)):
            while len(strs[i]) < len(prefix):
                prefix = prefix[:len(prefix) - 1]

            while strs[i][:len(prefix)] != prefix:
                prefix = prefix[:len(prefix) - 1]
                if not prefix:
                    return ""

        return prefix
